src/components/CreateManifestForm/CreateManifestForm.tsx

```diff
@@ -0,0 +1,234 @@
+import React, { useState, FC } from 'react';
+import {
+  Button,
+  ActionGroup,
+  Form,
+  FormGroup,
+  TextInput,
+  Title,
+  Popover,
+  FormSelect,
+  FormSelectOption
+} from '@patternfly/react-core';
+import ExclamationCircleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-circle-icon';
+import HelpIcon from '@patternfly/react-icons/dist/js/icons/help-icon';
+import CreateManifestFormError from './CreateManifestFormError';
+import CreateManifestFormLoading from './CreateManifestFormLoading';
+import CreateManifestFormSuccess from './CreateManifestFormSuccess';
+import { SatelliteVersion } from '../../hooks/useSatelliteVersions';
+
+interface CreateManifestFormProps {
+  satelliteVersions: SatelliteVersion[];
+  handleModalToggle: () => void;
+  submitForm: (name: string, version: string) => void;
+  isLoading: boolean;
+  isError: boolean;
+  isSuccess: boolean;
+}
+
+const CreateManifestForm: FC<CreateManifestFormProps> = (props) => {
+  const { satelliteVersions, handleModalToggle, submitForm, isLoading, isError, isSuccess } = props;
+
+  const [name, setName] = useState('');
+  const [version, setVersion] = useState('Select type');
+  const [nameValidated, setNameValidated] = useState('noval');
+  const [nameHasInvalidCharacters, setNameHasInvalidCharacters] = useState(false);
+  const [versionValidated, setVersionValidated] = useState('noval');
+
+  const clearErrors = () => {
+    setNameValidated('noval');
+    setVersionValidated('noval');
+    setNameHasInvalidCharacters(false);
+  };
+
+  const handleNameChange = (value: string) => {
+    if (name.length) {
+      setNameValidated('noval');
+    }
+    setName(value);
+  };
+
+  const handleSelectChange = (value: string) => {
+    setVersionValidated('noval');
+    setVersion(value);
+  };
+
+  const validateManifestName = (name: string) => {
+    if (name.length >= 100) {
+      return false;
+    }
+    if (name.length === 0) {
+      return true;
+    }
+
+    var regExp = new RegExp('^[0-9A-Za-z_.-]+$');
+    const result = regExp.test(name);
+    return regExp.test(name);
+  };
+
+  const validateForm = () => {
+    clearErrors();
+    let isValid = true;
+
+    if (!name.length) {
+      setNameValidated('error');
+      isValid = false;
+    }
+    if (validateManifestName(name) === false) {
+      setNameValidated('error');
+      setNameHasInvalidCharacters(true);
+    }
+
+    if (version === 'Select type') {
+      setVersionValidated('error');
+      isValid = false;
+    }
+    return isValid;
+  };
+
+  const handleFormSubmit = () => {
+    const formIsValid = validateForm();
```

Comment:
I feel kind of like we should be validating on blur or something, not just on submit.  It's especially apparent if I try to submit with no value for name, get the validation failure message, add a name, see the validation message clear, and then delete the name, and see that the validation message doesn't reappear.